codejail/tests/helpers.py

```diff
@@ -3,28 +3,85 @@
 """
 
 from contextlib import contextmanager
+import os.path
+import sys
+from unittest import SkipTest
+
+from codejail import jail
+from codejail import languages
 
-from codejail import jail_code
 
 SAME = object()
 
 
 @contextmanager
-def override_configuration(command, bin_path, user):
+def override_configuration(command, bin_path=SAME, user=SAME, lang=SAME):
     """
     Context manager to temporarily alter the configuration of a codejail
     command.
     """
-    old = jail_code.COMMANDS.get(command)
+    old = jail.COMMANDS.get(command)
     if bin_path is SAME:
-        bin_path = old['cmdline_start'][0]
+        bin_path = old.bin_path
     if user is SAME:
-        user = old['user']
+        user = old.user
+    if lang is SAME:
+        lang = old.lang
     try:
-        jail_code.configure(command, bin_path, user)
+        jail.configure(command, bin_path, user, lang)
         yield
     finally:
         if old is None:
-            del jail_code.COMMANDS[command]
+            del jail.COMMANDS[command]
+        else:
+            jail.COMMANDS[command] = old
+
+
+class JailMixin(object):
```

Comment:
I've always derived from unittest.TestCase for mixins like this, since it gives linters more information about the context they'll be used in.  I'm not sure if there are other considerations pushing toward `object`.